candle-transformers/src/models/stable_diffusion/utils.rs

```diff
@@ -13,3 +13,62 @@ pub fn linspace(start: f64, stop: f64, steps: usize) -> Result<Tensor> {
         Tensor::from_vec(vs, steps, &Device::Cpu)
     }
 }
+
+/// A linear interpolator for a sorted array of x and y values.
+struct LinearInterpolator<'x, 'y> {
+    xp: &'x Vec<f64>,
+    fp: &'y Vec<f64>,
+    cache: usize,
+}
+
+impl<'x, 'y> LinearInterpolator<'x, 'y> {
+    fn bsearch(&self, x: f64, idx_low: usize, idx_high: usize) -> usize {
```

Comment:
Could you add a comment on why `binary_search_by` or a variant cannot be used here?

Reply:
Good point, I changed to use partition_point.